rtl/eclrtl/rtldynfield.cpp

```diff
@@ -460,6 +612,56 @@ class CRtlFieldTypeDeserializer : public CInterfaceOf<IRtlFieldTypeDeserializer>
         }
         return info.createRtlTypeInfo();
     }
+
+    const RtlTypeInfo *deserializeType(MemoryBuffer &type)
+    {
+        FieldTypeInfoStruct info;
+        type.read(info.fieldType);
+        type.read(info.length);
+        if (info.fieldType & RFTMhasLocale)
+        {
+            StringBuffer s;
+            type.read(s);
+            info.locale = keep(s);
+        }
+        if (info.fieldType & RFTMhasChildType)
+        {
+            unsigned childIdx;
+            type.read(childIdx);
+            info.childType = lookupType(childIdx);
+        }
+        if (info.fieldType & RFTMhasFields)
+        {
+            unsigned numFields;
+            type.read(numFields);
+            info.fieldsArray = new const RtlFieldInfo * [numFields+1];
+            info.fieldsArray[numFields] = nullptr;
+            for (int n = 0; n < numFields; n++)
+            {
+                const char *fieldName;
+                type.read(fieldName);
```

Comment:
I don't think so - returns a pointer into the MemoryBuffer, not an allocated one.